staking_deposit/deposit.py

```diff
@@ -29,6 +30,17 @@ def check_python_version() -> None:
         sys.exit()
 
 
+def check_connectivity() -> None:
+    '''
+    Checks if there is an internet connection and warns the user if so.
+    '''
+    try:
+        urlopen('https://www.google.com/', timeout=2)
```

Comment:
I'm not a fan of this test here. It potentially tests too much (DNS, TCP, HTTPS, TLS and maybe a few other). For any quick replies, it would give a strong confidence that an Internet connection is present, but there is a non-significant chance of timeout and false negative.

I think I would prefer something like this DNS resolution:
```
import socket # up in the import section

socket.setdefaulttimeout(2)
socket.getaddrinfo('icann.org', 80)
```
`icann.org` being a domain name that is less likely to be blocked in Russia, China, Iran, Turkmenistan, Libya, and Sudan among others.

A DNS test is not perfect, but it would fail faster on a non-connected device. It would also reduce the false negatives.

Reply:
See https://docs.python.org/3/library/socket.html for the socket module documentation.

Reply:
I missed the socket callout in the issue notes.
Updated.